In [8]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [9]:
df = pd.read_csv('../data/photo_classify.csv')

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1822 entries, 0 to 1821
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   mach_learn  1822 non-null   object
 1   truth       1822 non-null   object
dtypes: object(2)
memory usage: 28.6+ KB


In [11]:
df.head()

mach_learn    truth
0  pred_fashion  fashion
1  pred_fashion  fashion
2  pred_fashion  fashion
3  pred_fashion  fashion
4  pred_fashion  fashion

### Marginal probabilities
    - Probability based on a single variables
    
### Joint Probabilities
    - Probability based on two or more variables

In [12]:
pd.crosstab(df['mach_learn'], df['truth'], margins=True)

truth         fashion   not   All
mach_learn                       
pred_fashion      197    22   219
pred_not          112  1491  1603
All               309  1513  1822

#### Marginal probability
$P(pred\text{_}fashion) = \frac{219}{1822}$

#### Joint probability
$P(pred\text{_}fashion \hspace{1mm} \text{and} fashion) = \frac{197}{1822}$

In [13]:
pd.crosstab(df['mach_learn'], df['truth'], margins=True, normalize='all')

truth          fashion       not       All
mach_learn                                
pred_fashion  0.108123  0.012075  0.120198
pred_not      0.061471  0.818332  0.879802
All           0.169594  0.830406  1.000000

### Conditional probability

$P(A | B) = \frac{P(A \hspace{1mm} \text{and} \hspace{1mm} B)}{P(B)}$

$P(\textit{truth is fashion} \hspace{1mm} \text{given} \hspace{1mm} \textit{mach_learn is pred_fashion}) = \frac{197}{219}$

$P(\textit{truth is fashion} \hspace{1mm} \text{|} \hspace{1mm} \textit{mach_learn is pred_fashion}) = \frac{197}{219}$

In [14]:
pd.crosstab(df['mach_learn'], df['truth'], margins=True, normalize='index')

truth          fashion       not
mach_learn                      
pred_fashion  0.899543  0.100457
pred_not      0.069869  0.930131
All           0.169594  0.830406

### General multiplication rule

$P(A \hspace{1mm} \text{and} \hspace{1mm} B) = P(A | B) \times {P(B)}$

### Bayes Theorem

$P(A_j | E) = \frac{P(E | A_j) P(A_j)}{\sum_{i} P(A_i)P(E | A_i)}$